test/demo/contextChannels.test.ts

```diff
@@ -0,0 +1,270 @@
+
+afterEach(async () => {
+    jest.clearAllMocks();
+
+    for (const app of startedApps) {
+        await app.quit().catch(() => {});
+    }
+
+    startedApps.length = 0;
+});
+
+// Creates one window for each channel, and has that window join that channel if not undefined
+async function setupWindows(...channels: (string|undefined)[]): Promise<Identity[]> {
+    const app1 = {uuid: 'test-app-1', name: 'test-app-1'};
+    const app2 = {uuid: 'test-app-2', name: 'test-app-2'};
+    const app3 = {uuid: 'test-app-3', name: 'test-app-3'};
+    const app4 = {uuid: 'test-app-4', name: 'test-app-4'};
+
+    const appIdentities = [app1, app2, app3, app4];
+
+    // Creating apps takes time, so increase timeout
+    jest.setTimeout(channels.length * 5000);
+
+    const result: Identity[] = await Promise.all(channels.map(async (channel, index) => {
+        const identity = appIdentities[index];
+
+        await fdc3Remote.open(testManagerIdentity, identity.uuid);
+        const app = fin.Application.wrapSync(appIdentities[index]);
+
+        await expect(app.isRunning()).resolves.toBe(true);
+
+        startedApps.push(app);
+        if (channel) {
+            await fdc3Remote.joinChannel(identity, channel);
+        }
+
+        return identity;
+    }));
+
+    return result;
+}
+
+describe('When broadcasting on global channel', () => {
+    it('Context is received by global windows only', async () => {
+        const [globalWindow, blueWindow] = await setupWindows(undefined, 'blue');
+
+        const globalListener = await fdc3Remote.addContextListener(globalWindow);
+        const blueListener = await fdc3Remote.addContextListener(blueWindow);
+
+        // Broadcast our context on the global channel
+        await fdc3Remote.broadcast(testManagerIdentity, testContext);
+
+        // Check the global window received our test context
+        const globalContexts = await globalListener.getReceivedContexts();
+        expect(globalContexts.length).toBe(1);
+        expect(globalContexts[0]).toEqual(testContext);
+
+        // Check the blue window received no context
+        const blueContexts = await blueListener.getReceivedContexts();
+        expect(blueContexts.length).toBe(0);
+    });
+
+    it('Context is received by window that has left and rejoined global channel', async () => {
+        const [channelChangingWindow] = await setupWindows('blue');
+
+        // Change the channel of our window
+        await fdc3Remote.joinChannel(channelChangingWindow, 'global');
+
+        const channelChangingListener = await fdc3Remote.addContextListener(channelChangingWindow);
+
+        // Broadcast our context on the global channel
+        await fdc3Remote.broadcast(testManagerIdentity, testContext);
+
+        // Check our window received our test context
+        const receivedContexts = await channelChangingListener.getReceivedContexts();
+        expect(receivedContexts.length).toBe(1);
+        expect(receivedContexts[0]).toEqual(testContext);
```

Comment:
Would `expect(receivedContexts).toEqual([testContext]);` be equivalent to both of these assertions?